okTurtles/group-income
# #893 Voting System - Add "Disagreement" rule

Closes #893

### What's done:
- A group can be created with a custom Voting System: _disagreement_ or _percentage_.
- The Voting System can be changed at Group Settings (directly or through a proposal)
- Cypress tests to make sure the Voting System is correct (in 6 different scenarios)
- Unit tests to the rule system. - Using Cypress to cover every single proposal scenario would increase the CI tests' time a lot. For that reason, I used unit tests instead for corner cases.

### How to test this:
There are a lot of possible scenarios based on the voting system config.

You can create multiple groups with different voting systems or create a single group and then change the voting system at the "Group Settings" page.
Here's a screenshot of how the Group Settings card looks like in 8 different scenarios:
**Note:** the last scenario is incorrect. it should say _adjusted to 4_, because the group has 5 members and the limit is `groupSize - 1`. The code was updated taking into account that.
![ss](https://user-images.githubusercontent.com/14869087/83394693-e9e81100-a3f0-11ea-99f8-969fafb3475b.png)

### Notes
- We have been naming this "Voting System", but the core code uses "Voting Rules". I changed all the Vue files names and props to follow the code "rule" naming.
- Some **layout decisions are still being discussed at Figma**. Once solved, I'll update this PR as well.
